drivers/misc/ethos_u/init.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2021-2022 Arm Limited and/or its affiliates <open-source-office@arm.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ethosu_driver.h"
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/sys/util.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ethos_u, CONFIG_ARM_ETHOS_U_LOG_LEVEL);
+
+#define DT_DRV_COMPAT arm_ethos_u
+
+/*******************************************************************************
+ * Re-implementation/Overrides __((weak)) symbol functions from ethosu_driver.c
+ * To handle mutex and semaphores
+ *******************************************************************************/
+
+void *ethosu_mutex_create(void)
+{
+	int status = 0;
+	struct k_mutex *mutex = k_malloc(sizeof(*mutex));
```

Comment:
The error handling in this function is not correct. k_malloc() is what can fail, not k_mutex_init(). This is missing a null check.

Reply:
Done.